### PR TITLE
build(root): Update npm packages

### PR DESCRIPTION
**Pull request by yasincaliskan**
### Description
- Update hipo/linter-packages
- Add husky & lint-staged packages
- Add npm scripts

### Note
- Merge after exist component PRs merged

### Todo
- Add `pre-push` husky script after test implementation

### Related Issue:
- #88

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,10 +7,18 @@
     "dev": "./node_modules/.bin/rollup -c -w",
     "build": "npm run prettier:fix && ./node_modules/.bin/rollup -c",
     "test": "jest",
```
Comment (Anlerkan) [inline package.json:9]: I think we can add `test:watch` script too

```suggestion
    "test": "jest",
    "test:watch":"npm t -- --watch",
```

Comment (yasincaliskan) [reply package.json:9]: I had missed that. Thank you!

File: package.json [hunk 2]
```diff
@@ -7,10 +7,19 @@
...
+    "type-check": "echo \"type-checking...\" && tsc --noEmit",
+    "type-check:watch": "npm run type-check -- --watch",
+    "eslint": "./node_modules/.bin/eslint --ext .jsx,.js,.tsx,.ts src/",
+    "eslint:fix": "./node_modules/.bin/eslint --fix --ext .jsx,.js,.tsx,.ts src/",
+    "stylelint": "./node_modules/.bin/stylelint \"src/**/*.{css,scss}\"",
+    "stylelint:fix": "npm run stylelint -- --fix",
+    "prettify": "./node_modules/.bin/prettier --config ./.prettierrc.js --write \"src/**/*.{ts,tsx}\" \"stories/**/*.{ts,tsx}\"",
+    "lint:fix": "npm run stylelint:fix && npm run eslint:fix",
+    "lint": "npm run stylelint && npm run eslint",
+    "checkPackageUpdates": "npx npm-check-updates"
```
Comment (mucahit) [inline package.json:22]:
```suggestion
    "check-package-updates": "npx npm-check-updates"
```

**Review by mucahit — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (yasincaliskan) [reply package.json:22]: I guess it the same on the [project-starter](https://github.com/Hipo/hipo-frontend-project-starter/blob/master/package.json). I can open a PR for this

**Review by Anlerkan — APPROVED** [review]

**Review by bilgeyucel — APPROVED** [review]

File: .eslintrc.js [hunk 3]
```diff
@@ -57,22 +56,9 @@ module.exports = {
...
-    "@typescript-eslint/explicit-member-accessibility": 0,
-    "@typescript-eslint/no-explicit-any": 0,
-    "@typescript-eslint/camelcase": 0,
-    "@typescript-eslint/ban-ts-ignore": 0,
-    "@typescript-eslint/no-non-null-assertion": 0,
-    "@typescript-eslint/explicit-module-boundary-types": 0,
     "@typescript-eslint/ban-ts-comment": 0,
-    "@typescript-eslint/no-empty-function": 0,
 
     // 👇🏻 these will be fixed inside `eslint-config-hipo-base`
```
Comment (jamcry) [inline .eslintrc.js:61]: I think we can remove this comment as well since they are fixed in the library

**Review by jamcry — APPROVED** [review]

**Review by edizcelik — APPROVED** [review]

**Pull request merged**